### PR TITLE
ci: add test coverage computation & upload

### LINKED ISSUE #8: Add coverage information
# Description

Add coverage information for PR and badges on the front page.

### PR DESCRIPTION
**Pull request by imrn99**
# Description

Add a new workflow to compute the test coverage of Rust code. The data is compiled into an `lcov` file & uploaded to codecov for hosting.

## Scope

- [x] CI

## Type of change

- [x] New feature(s)
- [x] Fix: closes #8 

## Other

...

## Necessary follow-up

...

## Mentions

...

### CODE REVIEW

File: .github/workflows/rust-test.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ jobs:
         with:
           submodules: recursive
       - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v2
```
Comment (cedricchevalier19) [inline .github/workflows/rust-test.yml:20]: Is this change related to coverage?

**Review by cedricchevalier19 — APPROVED** [review]

**Review by dssgabriel — APPROVED** [review]

File: .github/workflows/rust-test.yml (hunk 1 above)
Comment (imrn99) [reply .github/workflows/rust-test.yml:20]: It's not directly, but running the coverage action was taking quite some time. I added this to all files at the same time
